File: frontend/src/api/fertilizer.ts
import apiClient from './client';
import type { FertilizerPredictionRequest, FertilizerPredictionResponse } from '../types/fertilizer';

export const getFertilizerPrediction = async (data: FertilizerPredictionRequest): Promise<FertilizerPredictionResponse> => {
  const response = await apiClient.post('/fertilizer/predict', data);
  return response.data?.data || response.data;
};

export const getRecommendation = async (data: FertilizerPredictionRequest): Promise<FertilizerPredictionResponse> => {
  const response = await apiClient.post('/fertilizer/recommend', data);
  return response.data?.data || response.data;
};

File: backend/app/api/v1/fertilizer.py
from fastapi import APIRouter
from app.schemas.fertilizer import FertilizerPredictionRequest, FertilizerRequest
from app.schemas.common import APIResponse
from app.services.fertilizer_service import FertilizerService

router = APIRouter()

@router.post("/predict", response_model=APIResponse)
async def predict_fertilizer(data: FertilizerPredictionRequest):
    result = await FertilizerService.predict(data)
    return APIResponse(data=result)

@router.post("/recommend", response_model=APIResponse)
async def recommend_fertilizer(data: FertilizerRequest):
    result = await FertilizerService.recommend(data)
    return APIResponse(data=result)
